### PR TITLE
Remove EmitMetric option from the CLI #2380

### PR DESCRIPTION
**Pull request by luowenking1**
Instead of renaming the property which may call the back-end breaking, I removed it from CLI and mark the property as depreciated.

### CODE REVIEW

File: common/persistence/persistenceInterface.go [hunk 1]
```diff
@@ -599,7 +599,7 @@ type (
 	InternalDomainConfig struct {
 		// NOTE: this retention is in days, not in seconds
 		Retention                int32
-		EmitMetric               bool
+		EmitMetric               bool                    // deprecated
```
Comment (venkat1109) [inline common/persistence/persistenceInterface.go]: you can completely get rid of this. There is no concern with backward compatibility within the internal server code

File: tools/cli/flags.go [hunk 2]
```diff
@@ -123,8 +123,8 @@ const (
 	FlagOwnerEmailWithAlias               = FlagOwnerEmail + ", oe"
 	FlagRetentionDays                     = "retention"
 	FlagRetentionDaysWithAlias            = FlagRetentionDays + ", rd"
-	FlagEmitMetric                        = "emit_metric"
-	FlagEmitMetricWithAlias               = FlagEmitMetric + ", em"
+	FlagEmitMetric                        = "emit_metric"           // deprecated
```
Comment (venkat1109) [inline tools/cli/flags.go]: completely delete these. This is just internal constant

**Comment by luowenking1** [comment]
Pull request for task #2380

File: .gen/go/shared/shared.go [hunk 3]
```diff
@@ -33318,7 +33319,10 @@ func (v *RegisterDomainRequest) ToWire() (wire.Value, error) {
 		fields[i] = wire.Field{ID: 40, Value: w}
 		i++
 	}
-	if v.EmitMetric != nil {
+	if v.EmitMetric == nil {
```
Comment (venkat1109) [inline .gen/go/shared/shared.go]: I don't see the cadence-idl submodule update corresponding to this. Is that included ?

File: tools/cli/domainUtils.go [hunk 4]
```diff
@@ -119,10 +115,6 @@ var (
 			Name:  FlagRetentionDaysWithAlias,
 			Usage: "Workflow execution retention in days",
 		},
-		cli.StringFlag{
```
Comment (venkat1109) [inline tools/cli/domainUtils.go]: also please double check if this flag is currently being accessed in things like docker/scripts/buildkite etc. This is a breaking change i.e. will break scripts which already have this flag

File: .gen/go/shared/shared.go (hunk 3 above)
Comment (luowenking1) [reply .gen/go/shared/shared.go]: The idl update was committed to repo idl with this pr https://github.com/uber/cadence-idl/pull/10

Comment (luowenking1) [reply .gen/go/shared/shared.go]: > I don't see the cadence-idl submodule update corresponding to this. Is that included ?

The idl update was committed to repo idl with this pr uber/cadence-idl#10

File: tools/cli/app_test.go [hunk 5]
```diff
@@ -178,7 +178,7 @@ func (s *cliAppSuite) TestDomainUpdate() {
 	s.serverFrontendClient.EXPECT().UpdateDomain(gomock.Any(), gomock.Any()).Return(nil, nil).Times(2)
 	err := s.app.Run([]string{"", "--do", domainName, "domain", "update"})
 	s.Nil(err)
-	err = s.app.Run([]string{"", "--do", domainName, "domain", "update", "--desc", "another desc", "--oe", "another@uber.com", "--rd", "1", "--em", "f"})
+	err = s.app.Run([]string{"", "--do", domainName, "domain", "update", "--desc", "another desc", "--oe", "another@uber.com", "--rd", "1", "f"})
```
Comment (yycptt) [inline tools/cli/app_test.go]: also remove "f"?

Comment (luowenking1) [reply tools/cli/app_test.go]: I have removed it. Thanks.

File: tools/cli/domainUtils.go (hunk 4 above)
Comment (luowenking1) [reply tools/cli/domainUtils.go]: > also please double check if this flag is currently being accessed in things like docker/scripts/buildkite etc. This is a breaking change i.e. will break scripts which already have this flag

I have checked and resolved them in the unit tests.

**Pull request merged**